### PR TITLE
feat(cb2-11259): upgrade aws SDK to V3

### PR DESCRIPTION
**Pull request by JunYanBJSS**
## Ticket title
upgrade AWS SDK to V3
[CB2-11259](https://dvsa.atlassian.net/browse/CB2-11259)

## Checklist

- [ ] Branch is rebased against the latest develop
- [ ] PR title includes the JIRA ticket number
- [ ] Squashed commits contain the JIRA ticket number
- [ ] Link to the PR added to the repo

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -29,29 +29,32 @@
   "author": "",
   "license": "ISC",
   "dependencies": {
+    "@aws-sdk/client-dynamodb": "^3.540.0",
+    "@aws-sdk/client-s3": "^3.537.0",
```
Comment (naathanbrown) [inline package.json:33]: Where is this used? I can't see it? :(

File: src/functions/retroGenInit.ts [hunk 2]
```diff
@@ -1,9 +1,9 @@
 import { Callback, Context, DynamoDBRecord, Handler } from "aws-lambda";
```
Comment (naathanbrown) [inline src/functions/retroGenInit.ts:1]: This package is really old and would be great if we could move the types too

File: src/services/SQService.ts [hunk 3]
```diff
@@ -1,32 +1,41 @@
...
+  SendMessageCommandOutput,
+  SQS,
+  SQSClient,
+} from "@aws-sdk/client-sqs";
+import * as AWSXRay from 'aws-xray-sdk';
 import { Service } from "../models/injector/ServiceDecorator";
 import { Configuration } from "../utils/Configuration";
 import { PromiseResult } from "aws-sdk/lib/request";
-import { AWSError, config as AWSConfig } from "aws-sdk";
+import { config as AWSConfig } from "aws-sdk";
```
Comment (naathanbrown) [inline src/services/SQService.ts:20]: how are we importing from "aws-sdk" isn't that the v2 thing we are removing?

File: src/services/SQService.ts [hunk 4]
```diff
@@ -1,32 +1,41 @@
...
 /* tslint:disable */
-const AWSXRay = require("aws-xray-sdk");
 /* tslint:enable */
-
+// const AWSXRay = require('aws-xray-sdk');
```
Comment (naathanbrown) [inline src/services/SQService.ts:23]: think we need to leave this on?

**Review by naathanbrown — COMMENTED** [review]
Looks mostly good! check our slack messages before you start any fixes tho

File: tests/unit/retroGenInit.unitTest.ts [hunk 5]
```diff
@@ -1,10 +1,8 @@
 import { SQService } from "../../src/services/SQService";
```
Comment (cb-cs) [inline tests/unit/retroGenInit.unitTest.ts:1]: Not sure if anyone else is having the same issue, but I am having test failures when running locally after dep reinstall and project rebuild.

File: src/functions/retroGenInit.ts [hunk 6]
```diff
@@ -1,9 +1,9 @@
...
-import { AWSError } from "aws-sdk";
+import { ServiceException } from "@smithy/smithy-client";
+import { SendMessageCommandOutput } from "@aws-sdk/client-sqs";
 import { Injector } from "../models/injector/Injector";
 import { SQService } from "../services/SQService";
 import { PromiseResult } from "aws-sdk/lib/request";
```
Comment (cb-cs) [inline src/functions/retroGenInit.ts:6]: Unused import

File: src/services/SQService.ts [hunk 7]
```diff
@@ -1,32 +1,41 @@
-import SQS, {
-  GetQueueUrlResult,
-  MessageBodyAttributeMap,
-  ReceiveMessageResult,
-  SendMessageResult,
-} from "aws-sdk/clients/sqs";
+
+import { ServiceException } from "@smithy/smithy-client";
```
Comment (cb-cs) [inline src/services/SQService.ts:2]: Unused import and formatting

File: src/services/SQService.ts [hunk 8]
```diff
@@ -1,32 +1,41 @@
...
+import {
+  GetQueueUrlCommand,
+  GetQueueUrlCommandOutput,
+  MessageAttributeValue,
+  ReceiveMessageCommand,
+  ReceiveMessageCommandOutput,
+  SendMessageCommand,
+  SendMessageCommandInput,
+  SendMessageCommandOutput,
+  SQS,
```
Comment (cb-cs) [inline src/services/SQService.ts:13]: Unused import

File: src/services/SQService.ts [hunk 9]
```diff
@@ -64,29 +71,25 @@ class SQService {
...
       Object.assign(params, { MessageAttributes: messageAttributes });
     }
-
+    const sendMessage = new SendMessageCommand(params as SendMessageCommandInput);
     // Send a message to the queue
-    return this.sqsClient
-      .sendMessage(params as SQS.Types.SendMessageRequest)
-      .promise();
+    return await this.sqsClient.send(sendMessage);
+
```
Comment (cb-cs) [inline src/services/SQService.ts:77]: Can you look at the formatting here too please

File: src/services/SQService.ts [hunk 10]
```diff
@@ -64,29 +71,25 @@ class SQService {
...
+      
   }
 
   /**
    * Get the messages in the queue
    */
-  public async getMessages(): Promise<
-    PromiseResult<ReceiveMessageResult, AWSError>
-  > {
+  public async getMessages(): Promise<ReceiveMessageCommandOutput>
```
Comment (cb-cs) [inline src/services/SQService.ts:83]: Formatting too here please

File: tests/unit/retroGenInit.unitTest.ts [hunk 11]
```diff
@@ -30,48 +28,57 @@ describe("retro-gen-init", () => {
       () => {
         it("should result in an array of filtered js objects containing only visits", () => {
           processedEvent = StreamService.getVisitsStream(event as any);
+          console.log(processedEvent[0].id + '--------');
```
Comment (cb-cs) [inline tests/unit/retroGenInit.unitTest.ts:31]: Is the console log required here?

**Pull request closed without merging**